src/uwtools/tests/drivers/test_driver.py

```diff
@@ -108,46 +107,68 @@ def config(tmp_path):
 
 
 @fixture
-def assetobj(config):
-    return ConcreteAssets(
-        config=config,
-        dry_run=False,
-        batch=True,
-        cycle=dt.datetime(2024, 3, 22, 18),
-        leadtime=dt.timedelta(hours=24),
-    )
+def assetsobj(config):
+    return ConcreteAssetsTimeInvariant(config=config, dry_run=False)
 
 
 @fixture
 def driverobj(config):
-    return ConcreteDriver(
-        config=config,
-        dry_run=False,
-        batch=True,
-        cycle=dt.datetime(2024, 3, 22, 18),
-        leadtime=dt.timedelta(hours=24),
-    )
+    return ConcreteDriverTimeInvariant(config=config, dry_run=False, batch=True)
 
 
 # Assets Tests
 
 
-def test_Assets(assetobj):
-    assert Path(assetobj._driver_config["base_file"]).name == "base.yaml"
-    assert assetobj._batch is True
+def test_Assets(assetsobj):
+    assert Path(assetsobj._driver_config["base_file"]).name == "base.yaml"
+
 
+def test_Assets_repr_cycle_based(config):
+    obj = ConcreteAssetsCycleBased(config=config, cycle=dt.datetime(2024, 7, 2, 12))
+    expected = "concrete 2024-07-02T12:00 in %s" % obj._driver_config["run_dir"]
+    assert repr(obj) == expected
 
-@mark.parametrize("hours", [0, 24, 168])
-def test_Assets_cycle_leadtime_error(config, hours):
-    with raises(UWError) as e:
-        ConcreteAssets(config=config, leadtime=dt.timedelta(hours=hours))
-    assert "When leadtime is specified, cycle is required" in str(e)
+
+def test_Assets_repr_cycle_and_leadtime_based(config):
+    obj = ConcreteAssetsCycleAndLeadtimeBased(
+        config=config, cycle=dt.datetime(2024, 7, 2, 12), leadtime=dt.timedelta(hours=6)
+    )
+    expected = "concrete 2024-07-02T12:00 06:00:00 in %s" % obj._driver_config["run_dir"]
+    assert repr(obj) == expected
+
+
+def test_Assets_repr_time_invariant(config):
+    obj = ConcreteAssetsTimeInvariant(config=config)
+    expected = "concrete in %s" % obj._driver_config["run_dir"]
+    assert repr(obj) == expected
```

Comment:
The above three tests exercise the new, flexible `__repr__()` method defined in `Assets` and inherited by its subclass hierarchy.